FuelLabs/fuel-indexer
# refactor: swap ID from u64 to `SizedAsciiString<64>`

Thanks for opening a PR with the Fuel Indexer project, please review the **Checklist** below and ensure you've completed all of the necessary steps to make this PR review as painless as possible.


### Checklist
- [ ] Ensure your top-level commit message is inline with our [contributor guidelines](./CONTRIBUTING.md).
- [ ] Please allow Codeowners at least 24 hours to do a first-pass review.
- [ ] Please add thorougly detailed testing steps below.
- [ ] Please keep your Changelog message short and sweet.

### Description
- I noticed quite a bit of weirdness whilst working on this - so we won't be cutting 0.20 immediately until we add some more (relevant, non-redundant, useful) testing 
- PR does a few things
  - Does not optimize usage of `.clone()` https://github.com/FuelLabs/fuel-indexer/issues/1278
  - Refactors the project to use `SizedAsciiString<64>` IDs instead of `u64` IDs
    -  With `SizedAsciiString<64>` aliased as `UID`
  - Removes simple-wasm indexer
    - It wasn't testing anything we weren't already testing
  - Also includes more work from https://github.com/FuelLabs/fuel-indexer/issues/1194
    - We don't use any other DB anymore so removed "postgres" suffix from test modules 
    - Moves the fuel-indexer-test indexer from fuel-indexer-tests/components/indices/ to fuel-indexer-tests/indexers
      - simple-wasm too
    - Removes redundant tests (database) that have not detected much breakage in the past
    - Removes the graphql_schema test module
      - Our GraphQL layer is gonna get an entire makeover per https://github.com/FuelLabs/fuel-indexer/issues/1145
        - Grantees are using the SQL endpoint
   - Updates our trybuild tests (which have not been running on CI :/ )
     - Re-enables these on CI
   - Moves `IndexMetadataEntity` to `fuel_indexer_types`
     - And makes `fuel_indexer_lib` dependent on `fuel_indexer_types`
       - `fuel_indexer_types` is now the base crate (previously was `fuel_indexer_lib`)

### Testing steps

- As this was a very low-level/granular change, CI for everything should pass
- Start your indexer service on beta-4 `cargo run --bin fuel-indexer -- run --run-migrations --fuel-node-host beta-4.fuel.network --fuel-node-port 80`
- Deploy the default indexer `forc index new indexer-test --namespace fuellabs && cd indexer-test && forc index deploy`
- Let this index to around block 250K, you should have no warnings for constraint errors

### Changelog

- refactor: swap ID from u64 to `SizedAsciiString<64>`
